typescript.md

```diff
@@ -0,0 +1,73 @@
+TypeScript Code Style
+=====================
+
+## Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+- [General](#general)
+- [Naming](#naming)
+- [Typing](#typing)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## General
+
+This style guide is generally the same as the one for JavaScript.
+
+[&#8593; back to TOC](#table-of-contents)
+
+## Naming
+
+* `IInterfaceLikeThis` — notice `I` in the beginning.
```

Comment:
I used this style and style without beginning `I`, and now I have the following thoughts:

* Better if one part of the system depends from another using interfaces, not implementation. Thus, interfaces are met in code more often than implementations (classes). That's why I don't see why wee should add additional awkward prefix `I` for frequently names in code.
* Prefix `I` doesn't add clarity in signatures, because you actually always work with interface (some class implicit provides some interface). In important places interfaces can be easy detected by `implements` keyword.
* Popular existing TypeScript declarations (e.g. for `Node.js`, `express`) doesn't use `I` for interfaces. This can add mess in names in a project.
* In other names we don't use prefixes or suffixes (like `m_` for member or `g_` for global). Thus, this rule looks little inconsistently.
* For avoid name collision with interface and implementation I use the following rules:
   * Prefer add specific name, e.g. `ConsoleLogger implements Logger`.
   * Or just add `Impl` suffix: `PoolImpl implements Pool`.